packages/typespec-ts/tsconfig.json

```diff
@@ -5,8 +5,8 @@
     "composite": true,
     "lib": ["ES2019"],
     "target": "es2019",
-    "module": "NodeNext",
-    "moduleResolution": "NodeNext",
+    "module": "node16",
```

Comment:
The issue we found happens in dev version `"@azure-tools/typespec-azure-core": "0.35.0-dev.11"` and see pr https://github.com/Azure/autorest.typescript/pull/2053. And in our main we use the stable version not dev one.

I may suggest you to directly create a pr based on my branch. So we could verify it works or not.

Reply:
That change should happen regardless of the version of typespec that you are using. You might get into the same issue with other packages in the future. So I would just merge that if it is not causing any issues anyway. If there is more problem to the dev version then we can investigate further.